sky/setup_files/setup.py

```diff
@@ -20,17 +20,24 @@
 import re
 import subprocess
 import sys
-from typing import Dict, List
 
 import setuptools
 
+# __file__ is setup.py at the root of the repo. We shouldn't assume it's a
+# symlink - e.g. in the sdist it's resolved to a normal file.
 ROOT_DIR = os.path.dirname(__file__)
+SETUP_FILE_DIR = os.path.join(ROOT_DIR, 'sky', 'setup_files')
 INIT_FILE_PATH = os.path.join(ROOT_DIR, 'sky', '__init__.py')
 _COMMIT_FAILURE_MESSAGE = (
     'WARNING: SkyPilot fail to {verb} the commit hash in '
     f'{INIT_FILE_PATH!r} (SkyPilot can still be normally used): '
     '{error}')
 
+# setuptools does not include the script dir on the search path, so manually add
+# it so that we can import the dependencies file.
+sys.path.append(SETUP_FILE_DIR)
+import dependencies  # pylint: disable=wrong-import-position
```

Comment:
Is this a common practice?

Would it cause any issue with our wheel building during `sky launch`? Would be great if we can test it in a clean env, e.g. start a new container for running some tests.

Reply:
I tested the wheel build command manually as well as sky launch to make sure this would work. I looked for information on whether this is common practice but couldn't find any useful info.

Reply:
I found this: https://github.com/pypa/setuptools/discussions/3134
Obviously it's not ideal to hack sys.path. We could use importlib as suggested there. What do you think?

Reply:
python build systems are so complicated...

Reply:
Updated this to use runpy.run_path, which seems a lot cleaner.